crates/ra_project_model/src/lib.rs

```diff
@@ -24,6 +25,34 @@ pub use crate::{
     sysroot::Sysroot,
 };
 
+#[derive(Deserialize, Clone, Debug, PartialEq, Eq)]
+#[serde(rename_all = "lowercase")]
+pub enum InlayHintDisplayType {
+    Off,
+    TypeHints,
+    ParameterHints,
+    Full,
+}
+
+#[derive(Deserialize, Clone, Debug, PartialEq, Eq)]
+#[serde(rename_all = "camelCase", default)]
+pub struct InlayHintOptions {
+    pub display_type: InlayHintDisplayType,
+    pub max_length: Option<usize>,
+}
+
+impl InlayHintOptions {
+    pub fn new(max_length: Option<usize>) -> Self {
```

Comment:
Just a small nit: if I got it right, the `new` function is used in the test only, all but one cases are used as `new(None)` which is basically `InlayHintOptions::default()`, so we can remove this method, since it's not needed really.